site/components/map/Markers.tsx

```diff
@@ -0,0 +1,88 @@
+import { uniqueId } from "lodash";
+import { Dispatch, ReactElement, SetStateAction, memo } from "react";
+import { Marker } from "react-map-gl";
+import { Stop } from "../../schemas/consequence.schema";
+
+interface MarkersProps {
+    selected: Stop[];
+    searched: Stop[];
+    markerData: Stop[];
+    unselectMarker: (id: string) => void;
+    selectMarker: (id: string) => void;
+    handleMouseEnter: (id: string) => void;
+    setPopupInfo: Dispatch<SetStateAction<Partial<Stop>>>;
+}
+
+const Markers = ({
+    selected,
+    searched,
+    markerData,
+    unselectMarker,
+    selectMarker,
+    handleMouseEnter,
+    setPopupInfo,
+}: MarkersProps): ReactElement | null => {
+    const dataFromPolygon = markerData.filter((sToFilter: Stop) =>
```

Comment:
Could the empty `selected` array check be added outside the filter block so that it is evaluated once?

```suggestion
    const dataFromPolygon = markerData.filter((sToFilter: Stop) =>
      selected && selected.length > 0 ? !selected.map((s) => s.atcoCode).includes(sToFilter.atcoCode) : sToFilter,
    );
    
    const dataFromPolygon = selected && selected.length > 0 ? markerData.filter((sToFilter: Stop) =>
       !selected.map((s) => s.atcoCode).includes(sToFilter.atcoCode),
    ); : markerData
```